### PR TITLE
Fixes #37567 - Remove OVAL related permissions from DB

### PR DESCRIPTION
**Pull request by ofedoren**
@adamruzicka, could you please take a look?

### CODE REVIEW

**Comment by adamruzicka** [comment]
While you're at it, could you also purge the leftover "Run OVAL scans" job template?

**Comment by adamruzicka** [comment]
> The template itself was removed

But if it was seeded in the meantime, it will stick around, no?

**Comment by ofedoren** [comment]
> > The template itself was removed
> 
> But if it was seeded in the meantime, it will stick around, no?

Yes, you're right. Sorry, I kinda read it in a different way than you meant, fixed now.

File: db/migrate/20240617105409_remove_oval_permissions.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    Permission.where(resource_type: 'ForemanOpenscap::Cve').destroy_all
+
+    # Not a permission, but still a leftover from OVAL removal
+    if ForemanOpenscap.with_remote_execution?
+      rex_feature = RemoteExecutionFeature.find_by(label: 'foreman_openscap_run_oval_scans')
+      template_id = rex_feature.job_template_id
+      TemplateInvocation.where(template_id: template_id).delete_all
+      TemplateInput.where(template_id: template_id).delete_all
+      Template.find(template_id).delete
+      rex_feature.destroy
```
Comment (adamruzicka) [inline db/migrate/20240617105409_remove_oval_permissions.rb:14]: It seems the feature needs to go first

```
# foreman-rake db:migrate
== 20240617105409 RemoveOvalPermissions: migrating ============================
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::ForeignKeyViolation: ERROR:  update or delete on table "templates" violates foreign key constraint "fk_rails_62d06fbd3e" on table "remote_execution_features"
DETAIL:  Key (id)=(216) is still referenced from table "remote_execution_features".
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/postgresql_adapter.rb:672:in `exec_params'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/postgresql_adapter.rb:672:in `block (2 levels) in exec_no_cache'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/postgresql_adapter.rb:671:in `block in exec_no_cache'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract_adapter.rb:696:in `block (2 levels) in log'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract_adapter.rb:695:in `block in log'
/usr/share/gems/gems/activesupport-6.1.7.6/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract_adapter.rb:687:in `log'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/postgresql_adapter.rb:670:in `exec_no_cache'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/postgresql_adapter.rb:649:in `execute_and_clear'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/postgresql/database_statements.rb:72:in `exec_delete'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract/database_statements.rb:185:in `delete'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/connection_adapters/abstract/query_cache.rb:22:in `delete'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/persistence.rb:395:in `_delete_record'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/persistence.rb:877:in `_delete_row'
/usr/share/gems/gems/activerecord-6.1.7.6/lib/active_record/persistence.rb:523:in `delete'
/usr/share/gems/gems/foreman_openscap-7.1.1/db/migrate/20240617105409_remove_oval_permissions.rb:13:in `up'
```

**Review by adamruzicka — CHANGES_REQUESTED** [review]

File: db/migrate/20240617105409_remove_oval_permissions.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class RemoveOvalPermissions < ActiveRecord::Migration[6.1]
+  def up
+    Permission.where(resource_type: 'ForemanOpenscap::OvalPolicy').destroy_all
+    Permission.where(resource_type: 'ForemanOpenscap::OvalContent').destroy_all
+    Permission.where(resource_type: 'ForemanOpenscap::Cve').destroy_all
+
+    # Not a permission, but still a leftover from OVAL removal
+    if ForemanOpenscap.with_remote_execution?
+      rex_feature = RemoteExecutionFeature.find_by(label: 'foreman_openscap_run_oval_scans')
+      template_id = rex_feature.job_template_id
```
Comment (adamruzicka) [inline db/migrate/20240617105409_remove_oval_permissions.rb:10]: Also, what if the user re-mapped the feature to a different template? Shouldn't we rather remove the one we seeded?

Comment (ofedoren) [reply db/migrate/20240617105409_remove_oval_permissions.rb]: I'd hope that no one did that. Even though they did, we won't support that in any way. I can imagine they might want to use that template to run some jobs manually with direct call to the `oscap` utility though... Thanks, updated.

File: db/migrate/20240617105409_remove_oval_permissions.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class RemoveOvalPermissions < ActiveRecord::Migration[6.1]
+  def up
+    Permission.where(resource_type: 'ForemanOpenscap::OvalPolicy').destroy_all
+    Permission.where(resource_type: 'ForemanOpenscap::OvalContent').destroy_all
+    Permission.where(resource_type: 'ForemanOpenscap::Cve').destroy_all
```
Comment (adamruzicka) [inline db/migrate/20240617105409_remove_oval_permissions.rb:5]: Not sure why anyone would do that, but would this work if anyone has a filter set up that uses any of these permissions? Iirc in that case we may need to drop things from the filterings table first.

**Comment by ofedoren** [comment]
Thanks, @adamruzicka, should be fixed now.

**Review by adamruzicka — APPROVED** [review]
LGTM

**Pull request merged**